torch/_dynamo/allowed_functions.py

```diff
@@ -54,40 +54,71 @@ def make_function_id_set(lazy_initializer):
     added to the graph and what will cause a graph break.
     """
 
-    class FunctionIdSet:
-        function_ids: Optional[Set[int]] = None
-        function_names: Optional[Dict[int, str]] = None
-
-        def __call__(self):
-            if self.function_ids is None:
-                value = lazy_initializer()
-                if isinstance(value, dict):
-                    self.function_ids = set(value.keys())
-                    self.function_names = value
+    _pending_functions: List[Tuple["FunctionIdSet", bool, int]] = []
+    _function_ids: Optional[Set[int]] = None
+    _function_names: Optional[Dict[int, str]] = None
+
+    def __init__(self, lazy_initializer) -> None:
+        self.lazy_initializer = lazy_initializer
+
+    def _initialize(self) -> None:
```

Comment:
Why do we need both `_initialize` and `_resolve`?  It seems like some methods call `_initialize` and others call `_resolve`.  What invariant are we using to ensure it is fully initialized?

Reply:
`_initialize()` gets the list initialized, but doesn't resolve any pending operations. For convenience, we are initializing the internal list when we access `function_ids` and `function_names` attributes, to make code more readable. Pending operations are to be resolved at the time we do any lookup in any of the lists. This is the reason `_resolve()` is only called during `__call__()`.